src/storage/redis_metadata.h

```diff
@@ -335,3 +336,15 @@ class HyperLogLogMetadata : public Metadata {
 
   EncodeType encode_type = EncodeType::DENSE;
 };
+
+class CountMinSketchMetadata : public Metadata {
+ public:
+  uint32_t width;
+  uint32_t depth;
+  uint64_t counter = 0;
+  std::vector<uint32_t> array;
```

Comment:
@git-hulk @PragmaTwice 

I'm thinking about handling the `array`. Since regarding the cmsketch as a string is ok to me, but parsing it to an `std::vector` is a bit weird to me. Can we take a `Buffer` and do zero copying when doing this?

Like:
```
LoadMetadata from string. When storing, forcing LittleEndian in the array
When reading from metadata, report error if not length enough, and hold a sliced buffer on underlying data
When read/write, using memcpy to avoid unaligned access.
```

Reply:
Yeah it can be like how we handle the JSON data structure.

Reply:
But here I'm more interested in whether it's better to be a single key or putting the array in subkeys.

Reply:
> But here I'm more interested in whether it's better to be a single key or putting the array in subkeys.

Actually I think this in single key is ok since we merely "only query metadata" if not calling "info"

Reply:
What's the limit of the size of such an array? Do we need to consider to split it?

Reply:
https://github.com/apache/kvrocks/pull/2524/files#r1779978212

@PragmaTwice The problem is that, the stored value would always be fetched together when required. It's merely to "partition querying" / "updating" since it's a prob structure

We can limit it to 50M or less?

Reply:
I don't know if it's good to have one rocksdb key with a 50M-bytes value.

I checked the code and seems we'll access `d` elements (once for every row of `w` elements) in the `d*w` array.
Then I think maybe it's good to keep it in one key if the total size is small enough, e.g. 1M.

And if it's larger, we can move them into `w/k` key-values (one column as one key), and calculate how many keys we need to retrieve.

Reply:
> I don't know if it's good to have one rocksdb key with a 50M-bytes value.

The default engine doesn't handle this well. There're some blog engine which could do this, and we can enable the blob in kvrocks here.

Perhaps we can limit the size to 1MB firstly? 🤔

Reply:
Looks good to me.

Also yeah we should not put the array inside the metadata.

Reply:
I may try a round tonight

Reply:
I'll adjust it to 1MB limit